Source/UI/Controllers/CardViewController.swift

```diff
@@ -451,13 +469,55 @@ public class CardViewController: UIViewController,
 
 extension CardViewController: BillingFormViewModelDelegate {
 
-    func onTapDoneButton(data: BillingForm) {
+    private func updateBillingFormDetailsInputView(data: BillingForm?) {
+        guard var summaryCellButtonStyle = summaryCellButtonStyle else { return }
+        guard let data = data else { return }
+        guard let address = data.address, let phone = data.phone else {
+            let style = DefaultPaymentSummaryEmptyDetailsCellStyle()
+            cardView.updateBillingFormEmptyDetailsInputView(style: style)
+            return
+        }
+        var summaryValue = ""
+        updateSummaryValue(with: data.name)
+        updateSummaryValue(with: address.addressLine1)
+        updateSummaryValue(with: address.addressLine2)
+        updateSummaryValue(with: address.city)
+        updateSummaryValue(with: address.state)
+        updateSummaryValue(with: address.zip)
+        updateSummaryValue(with: address.country?.name)
+        updateSummaryValue(with: phone.number, withNewLine: false)
+        summaryCellButtonStyle.summary.text = summaryValue
+        cardView.updateBillingFormSummaryView(style: summaryCellButtonStyle)
+
+        func updateSummaryValue(with value: String?, withNewLine: Bool = true) {
```

Comment:
updateSummaryValue can be outside the function its a function within a function ?. Also can we add some unit tests for this. Its a valid candidate for unit tests.